core/src/main/scala/cats/data/NonEmptyVector.scala

```diff
@@ -223,6 +223,8 @@ object NonEmptyVector extends NonEmptyVectorInstances {
     new NonEmptyVector(buf.result)
   }
 
+  def unapply[A](nev: NonEmptyVector[A]): Some[(A, Vector[A])] = Some((nev.head, nev.tail))
```

Comment:
Is there a foreseeable use-case where having the static return type as `Some` instead of `Option` will cause inference (or similar) issues?